### ISSUE TITLE
Set Non-Random Teams in PVP (F.R.)

### ISSUE THREAD

**Issue by Ek-Videogames** [Regalis11/Barotrauma#3547]
In sub-vs-sub / search-and-destroy servers, some setting to switch between either 
random teams , host decides teams , players decide teams. 
Possibly an additional second option weather or not to enforce team-balancing.

**Comment by EtiamNullam** [comment]
I agree. I really could use setting my own teams the last time I tried versus mode.

**Comment by Regalis11** [comment]
Partially addressed in https://github.com/Regalis11/Barotrauma-development/issues/2217

**Comment by JeroonBosch** [comment]
Current implementation allows only for players to set a team preference. The teams are only actually assigned after round start.
![image](https://github.com/FakeFishGames/Barotrauma/assets/8641689/eb68ba04-4674-4a82-a5fb-19e7adcdedba)

TODO needs mockup and design on how to actually select a team.

**Comment by JeroonBosch** [comment]
![image](https://github.com/FakeFishGames/Barotrauma/assets/8641689/d2cf1751-f30a-476b-bad1-fcdcaa8e14bd)

Team selection mode ("The method at which teams are formed in PvP mode" on-hover):
- Player preference (current functionality)
- Player chosen (functionality described below)

Auto-balance teams threshold ("Ensure teams are equal before starting a round). Values:
- 0 (off)
- 1 (default value)
- 2
- 3

**Choose team**
Replace "No preference" with "Spectate" or "No team" - the value/variable is directly linked to the Spectator checkbox. When a team is selected, "Spectate" checkbox is no longer checked.

If auto-balance is on, grey out and make it unable to join a team that's already at it's limit.

Upon certain triggers (1. when a player leaves a server, 2. when a player leaves a team to join spectators, maybe more) reevaluate auto-balance and do a countdown in chat:

> Server: Teams are imbalanced, automatically balancing the teams in 10 seconds...

10 seconds later it will switch random player(s) part of the full team to the less full team

> Server: Teams have been balanced automatically. Switched players: [player A], [Player B]

**Comment by JeroonBosch** [comment]
Team selection modes that can be considered in the future:
- Force random (functions as if all players have "no preference" selected)
- Server assigned (server gives all players a team, players cannot switch.. probably needs more thinking and imo we can skip this and better allow players to police it with "Player chosen")

**Comment by Rokvach** [comment]
Tested, working correctly. Can be merged and closed.
